filebeat/input/filestream/fswatch_test.go

```diff
@@ -118,6 +119,58 @@ func setupFilesForScannerTest(t *testing.T, tmpDir string) {
 	}
 }
 
+func TestFileWatcherRenamedTruncated(t *testing.T) {
+	tmpDir := t.TempDir()
+
+	fs, err := newFileScanner([]string{filepath.Join(tmpDir, "app.log*")}, fileScannerConfig{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	w := fileWatcher{
+		interval:     1 * time.Second,
+		log:          logp.L(),
+		scanner:      fs,
+		events:       make(chan loginp.FSEvent),
+		sameFileFunc: os.SameFile,
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go w.Run(ctx)
+
+	appLogPath := filepath.Join(tmpDir, "app.log")
+	rotatedAppLogPath := filepath.Join(tmpDir, "app.log.1")
+	err = ioutil.WriteFile(appLogPath, []byte("my longer log line"), 0o600)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	evt := w.Event()
+	require.Equal(t, evt.Op, loginp.OpCreate, "new file should be detected")
+	require.Equal(t, evt.OldPath, "", "new file does not have an old path set")
+	require.Equal(t, evt.NewPath, appLogPath, "new file does not have an old path set")
```

Comment:
It's a detail, but the signature of [`requre.Equal`](https://pkg.go.dev/github.com/stretchr/testify/require#Equal) is:

```
func Equal(t TestingT, expected interface{}, actual interface{}, msgAndArgs ...interface{})
```

The expected value comes first. This inversion will make the test error harder to understand.
```suggestion
	require.Equal(t, loginp.OpCreate, evt.Op, "new file should be detected")
	require.Equal(t, "", evt.OldPath, "new file does not have an old path set")
	require.Equal(t, appLogPath, evt.NewPath, "new file does not have an old path set")
```

The same applies to the other similar blocks on this function.